### ISSUE TITLE
Bug: Docker: The user limit on the total number of inotify watches has been reached

### ISSUE THREAD

**Issue by antonyh** [abraunegg/onedrive#2210]
### Describe the bug

Inside Docker, we get an error

```
The user limit on the total number of inotify watches has been reached.
To see the current max number of watches run:
sysctl fs.inotify.max_user_watches
To change the current max number of watches to 524288 run:
sudo sysctl fs.inotify.max_user_watches=524288
```

This is not the same as #962 because unless I'm mistaken this should be set in the Docker image and isn't controlled by the host OS.

### Operating System Details

```shell
Docker version 20.10.5+dfsg1, build 55c4c88
on Debian 11
Linux 5.10.0-19-amd64 #1 SMP Debian 5.10.149-2 (2022-10-21) x86_64 GNU/Linux
```

### Client Installation Method

From 3rd Party Source (PPA, OpenSuSE Build Service etc)

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

Docker 2.4-alpine

### What is your OneDrive Application Configuration

```shell
None/default
```

### What is your 'curl' version

```shell
As provided in the 2.4-alpine docker image.
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
Managed by docker
```

### What are all your local file system partition types

```shell
Managed by docker
```

### How do you use 'onedrive'

Folder is used by Docker Onedrive only for sync. 

### Steps to reproduce the behaviour

Using docker-compose:

```
$ cat docker-compose.yml
version: '3.1'

services:
    onedrive:
        image: driveone/onedrive:2.4-debian
        restart: unless-stopped
        stdin_open: true
        tty: true
        environment:
                - ONEDRIVE_UID=1000
                - ONEDRIVE_GID=1000
        volumes:
                - ${HOME}/OneDriveConfig:/onedrive/conf
                - ${HOME}/OneDrive:/onedrive/data
```

Sync starts after login, then error message is displayed via `docker logs <container_id>`

### Complete Verbose Log Output

```shell
Can be provided if needed but I think this is a docker image config issue.
```

### Additional context

Clean install, first sync download.

Also tried 'latest' but that's 4 months old now. It seems to work, albeit slowly. 

Also tried '2.4-debian' with same result.

I need a stable release, which rules out 'edge' builds.

**Comment by antonyh** [comment]
On the offchance running sudo sysctl fs.inotify.max_user_watches on the host would fix it, I tried. It does. Sorry for the phantom issue report.

**Comment by abraunegg** [comment]
@antonyh 
> Also tried 'latest' but that's 4 months old now.

This is because the build process for v2.4.21 encountered an error due to GitHub build process failing - please read: https://github.com/abraunegg/onedrive/discussions/2163 for details. This will be corrected when v2.4.22 is released and not before due to issues with qemu and memory allocation to build the Docker containers.

> I need a stable release, which rules out 'edge' builds

'Edge' builds are just-as-stable as the 'release' builds .... there is nothing 'unstable' about them.

> This is not the same as https://github.com/abraunegg/onedrive/issues/962 because unless I'm mistaken this should be set in the Docker image and isn't controlled by the host OS.

Regarding your error - essentially it is the *same* issue - inside the Docker container or not - the 'default' Linux inotify watches for your chosen distribution is set at an arbitrary level. When this limit is reached, this is when this error occurs.

To know what value to use, so that you dont run into the issue is to follow this advice: https://github.com/abraunegg/onedrive/issues/962#issuecomment-647969737

The value (524288) provided in the application output is just an *example* ... it can be used, but if you have more files than this, you will encounter the same issue. This is also explained here: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-in-monitor-mode

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
